packages/quantic/force-app/main/default/lwc/quanticFeedbackModal/quanticFeedbackModal.js

```diff
@@ -0,0 +1,234 @@
+import yourResponsesHelp from '@salesforce/label/c.quantic_YourResponsesHelp';
+import {getBueno} from 'c/quanticHeadlessLoader';
+import LightningModal from 'lightning/modal';
+import {api} from 'lwc';
+import errorTemplate from './error.html';
+import feedbackFormTemplate from './feedbackForm.html';
+import successTemplate from './success.html';
+
+/**
+ * The `QuanticFeedbackModal` component overlays a message modal on top of the current app window, the modal contains a form that allows the user to give feedback.
+ *
+ * @category Search
+ * @category Insight Panel
+ */
+export default class QuanticFeedbackModal extends LightningModal {
+  /**
+   * @api
+   * The list of options to be displayed in the modal.
+   * @type {Array<{label: string, value: string, withDetails: boolean, detailsRequired: boolean}>}
+   */
+  @api options;
+  /**
+   * @api
+   * The function that will be executed when the feedback is submitted.
+   * @type {function}
+   */
+  @api submitFeedback;
+
+  labels = {
+    done,
+    cancel,
+    sendFeedback,
+    explainWhy,
+    selectOneOfOptions,
+    selectTheReason,
+    whatWasTheReason,
+    fillOutField,
+    yourResponsesHelp,
+    thankYouForYourFeedback,
+  };
+
+  /** @type {string} */
+  feedbackValue;
+  /** @type {string} */
+  detailsValue;
+  /** @type {boolean} */
+  feedbackSubmitted = false;
+  /** @type {boolean} */
+  detailsInputError = false;
+  /** @type {string} */
+  error;
+  /** @type {boolean} */
+  validated = false;
+
+  connectedCallback() {
+    this.validateFeedbackModal();
+  }
+
+  /**
+   * Validates the properties provided to the feedback modal.
+   */
+  validateFeedbackModal() {
+    getBueno(this).then(() => {
+      this.validateTheOptionsProperty();
+      if (!this.error) {
+        this.validateOptions();
+      }
+      if (!this.error) {
+        this.validateTheSubmitFeedbackProperty();
+      }
+      this.validated = true;
+    });
+  }
+
+  validateTheOptionsProperty() {
+    if (!Bueno.isArray(this.options)) {
+      console.error('The options provided are not in a valid array.');
+      this.setError();
+    } else if (!this.options.length) {
+      console.error('At least one option must be specified.');
+      this.setError();
+    }
+  }
+
+  validateOptions() {
+    const missingValueOrLabel = this.options?.some(
+      ({value, label}) => !value || !label
+    );
+    if (missingValueOrLabel) {
+      console.error(
+        'Each option requires a label and a value to be specified.'
+      );
+      this.setError();
+    } else {
+      this.options.forEach(({value, label}) => {
+        if (!Bueno.isString(value)) {
+          console.error(`The "${value}" value is not a valid string.`);
+          this.setError();
+        }
+        if (!Bueno.isString(label)) {
+          console.error(`The "${label}" label is not a valid string.`);
+          this.setError();
+        }
+      });
+    }
```

Comment:
Yea I don't think it's necessary to validate all of that in this component.
Look at the lightning `datatable` component, it also receives options as part of a structured object.
I think we should just make sure that we are as forgiving as possible when stuff is missing. For example in the case of the data table, if you omit the label of a field, it just doesn't displays a label (but also doesn't just displays "undefined")

![image](https://user-images.githubusercontent.com/2488155/220179048-94cff957-7841-4c71-989e-fcb5d8f9f00b.png)

I feel that otherwise all this validation is adding a lot of code that will mostly just trip the user when they are making their component/code, but provide little to no value to the user.

Reply:
After offline discussion, later down the road we might move all this validation to a single centralized files, but for now it'll remain in this file.